source/dotnet/Services/Infrastructure/Persistence/Outbox/IOutboxMessageRepository.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Energinet.DataHub.Wholesale.Domain;
+using NodaTime;
+
+namespace Energinet.DataHub.Wholesale.Infrastructure.Persistence.Outbox;
+
+public interface IOutboxMessageRepository
+{
+    Task AddAsync(OutboxMessage message, CancellationToken token);
+
+    Task<IList<OutboxMessage>> GetByTakeAsync(int numberOfElements, CancellationToken token);
+
+    void DeleteByCreationDate(Instant date);
```

Comment:
This sounds like a method that will delete all messages that were created on the specified date.
Also, you should care more about whether the messages has been completed or not.
How about something like this?
```suggestion
    void DeleteProcessedOlderThan(TimeSpan timeSpan);
```

Reply:
Funny. I called it that last week. Changed.